src/bsoncxx/CMakeLists.txt

```diff
@@ -170,12 +170,14 @@ install(FILES
     COMPONENT dev
 )
 
-install(TARGETS
-    bsoncxx
-    RUNTIME DESTINATION bin COMPONENT runtime
-    LIBRARY DESTINATION lib COMPONENT runtime
-    ARCHIVE DESTINATION lib COMPONENT dev
-)
+if (BUILD_SHARED_LIBS)
+    install(TARGETS
+        bsoncxx
+        RUNTIME DESTINATION bin COMPONENT runtime
+        LIBRARY DESTINATION lib COMPONENT runtime
+        ARCHIVE DESTINATION lib COMPONENT dev
```

Comment:
See https://cmake.org/cmake/help/v3.2/command/install.html#installing-targets

For DLL platforms you need RUNTIME and ARCHIVE, for non-dll platforms you need LIBRARY.